dht_test.go

```diff
@@ -788,7 +788,7 @@ func TestRefresh(t *testing.T) {
 		}
 	}()
 
-	waitForWellFormedTables(t, dhts, 7, 10, 20*time.Second)
```

Comment:
Hey. I think the test was timing out(> 10 minutes) because of heavy lock contention on the routing table lock due to frequently polling the size in `waitForWellFormedTables `(polling interval is just 5ms) and accessing it as part of the bootstrap query.

I've refactored this test to make it work. Lmk if it makes sense.

Reply:
I guess this is fine, but it's a little strange. Also this test now must take 5 seconds since we have to wait for the context to timeout. Probably would be better to refactor `waitForWellFormedTables`  into a second `checkWellFormedTables`  function that we can call here every 50ms. Could do this in a followup PR though.